internal/provider/resource_ably_namespace.go

```diff
@@ -0,0 +1,289 @@
+package ably_control
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	ably_control_go "github.com/ably/ably-control-go"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type resourceNamespaceType struct{}
+
+// Get Namespace Resource schema
+func (r resourceNamespaceType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Attributes: map[string]tfsdk.Attribute{
+			"app_id": {
+				Type:        types.StringType,
+				Required:    true,
+				Description: "The application ID.",
+			},
+			"id": {
+				Type:        types.StringType,
+				Required:    true,
+				Description: "The namespace or channel name that the channel rule will apply to.",
```

Comment:
Add ForceNew to recreate the resource if `id` param changes.

Reply:
https://www.terraform.io/plugin/sdkv2/schemas/schema-behaviors#forcenew

Reply:
Scrap that - The link and param I provided were for TF SDK v2. For the Plugin Framework, use RequiresReplace param - https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/tfsdk#RequiresReplace

Reply:
Docs explaining further here - https://www.terraform.io/plugin/framework/resources/plan-modification